server/app/Icepeak.hs

```diff
@@ -86,15 +73,17 @@ main = do
   installHandlers core serv
   logAuthSettings config (coreLogRecords core)
   logQueueSettings config (coreLogRecords core)
+  logSyncSettings config (coreLogRecords core)
   log "System online. ** robot sounds **" (coreLogRecords core)
 
   -- TODO: Log exceptions properly (i.e. non-interleaved)
   void $ Async.wait pops
   void $ Async.wait upds
   void $ Async.wait serv
   void $ Async.wait logger
-  -- kill metrics endpoint when the main threads ended
+  -- kill auxiliary threads when the main threads ended
   Async.cancel metrics
+  Async.cancel sync
```

Comment:
```
  void $ Async.wait pops
```
This is the core loop that is stopped when `Ctrl+C`ing by posting the `Stop` command https://github.com/channable/icepeak/blob/9b296d87b4f4afad0208594409296576966ca8a2/server/app/Icepeak.hs#L36. Therefore, in main we simply wait until it's time to stop.

```
  void $ Async.wait upds
```
This is the websocket loop, which receives it's cancellation via the subscription queue from the queue, therefore we also just need to wait until it's done.

```
  void $ Async.wait serv
```
The `Ctrl+C` handler also `Async.cancel`s the server thread (because any requests coming in afterwards would have no effect, so it needs to be stopped at the same time as the core loop).

```
  void $ Async.wait logger
```
The logger is again stopped by the `Core`.

```
  Async.cancel metrics
```
The metrics server would run indefinitely, but since the core functionality of the application is done here already, we have to stop it as well.

```
  Async.cancel sync
```
The sync thread only has a one-way channel of communicating with the core, therefore it doesn't get a notification when the `Core` stops and therefore must be canceled explicitly.